server/middleware/auth.ts

```diff
@@ -0,0 +1,31 @@
+import { RequestHandler } from "express";
+import { verify } from "jsonwebtoken";
+
+import { User } from "../models";
+import { DecodedDataType } from "../interfaces";
+
+const auth: RequestHandler = async (req, res, next) => {
+    if(req.method === "OPTIONS") return next();
+
+    try {
+        const header = req.header("Authorization");
+        if(!header) throw new Error("Authorization header is absent");
```

Comment:
It's a good feature to don't write `{}`. But to. be better (for code reading) if you add braces :)